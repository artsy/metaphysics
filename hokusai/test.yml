version: "2"
services:
  metaphysics:
    command: env NODE_OPTIONS=--max_old_space_size=4096 yarn ci
    extends:
      file: build.yml
      service: metaphysics
    environment:
      - MEMCACHED_URL=metaphysics-memcached:11211
      - CI_PULL_REQUEST=$CI_PULL_REQUEST
      - CIRCLE_PULL_REQUEST=$CIRCLE_PULL_REQUEST
      - CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM
      - ENABLE_CONSIGNMENTS_STITCHING=true
    env_file: ../.env.test
    depends_on:
      - metaphysics-memcached
  metaphysics-memcached:
    image: memcached
    ports:
      - 11211:11211
