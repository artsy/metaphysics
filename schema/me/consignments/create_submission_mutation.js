// @ts-check

import { mutationWithClientMutationId } from "graphql-relay"
import { SubmissionType } from "./submission"
import { omit } from "lodash"

export const config = {
  name: "CreateSubmissionMutation",
  description: "Create a new consignment submission using Convection",
  inputFields: {
    ...omit(SubmissionType.getFields(), ["id"]),
  },
  outputFields: {
    submission: {
      type: SubmissionType,
      resolve: response => response,
    },
  },
  mutateAndGetPayload: (request, _response, { rootValue: { submissionCreateLoader } }) => {
    if (!submissionCreateLoader) return null
    return submissionCreateLoader(request)
  },
}
export default mutationWithClientMutationId(config)
