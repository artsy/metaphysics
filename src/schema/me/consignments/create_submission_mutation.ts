import { mutationWithClientMutationId, MutationConfig } from "graphql-relay"
import { SubmissionType } from "./submission"
import { omit } from "lodash"
import { ResolverContext } from "types/graphql"

export const config: MutationConfig<any, any, ResolverContext> = {
  name: "CreateSubmissionMutation",
  description: "Create a new consignment submission using Convection",
  inputFields: {
    ...omit(SubmissionType.getFields(), ["id", "_id", "__id", "artist"]),
  } as any,
  outputFields: {
    consignment_submission: {
      type: SubmissionType,
      resolve: response => response,
    },
  },
  mutateAndGetPayload: (request, { submissionCreateLoader }) => {
    if (!submissionCreateLoader) return null
    return submissionCreateLoader(request)
  },
}
export default mutationWithClientMutationId(config)
