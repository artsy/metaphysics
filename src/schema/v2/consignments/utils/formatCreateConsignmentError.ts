import {
  GraphQLInt,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLString,
} from "graphql"
import { formatGravityError } from "lib/gravityErrorHandler"
import { ResolverContext } from "types/graphql"

export const ConsignmentInquiryMutationErrorName =
  "ConsignmentInquiryMutationError"

export const ConsignmentInquiryErrorType = new GraphQLObjectType<
  any,
  ResolverContext
>({
  name: ConsignmentInquiryMutationErrorName,
  fields: () => ({
    type: {
      type: GraphQLString,
    },
    message: {
      type: new GraphQLNonNull(GraphQLString),
    },
    error: {
      type: GraphQLString,
    },
    statusCode: {
      type: GraphQLInt,
    },
  }),
})

export const formatCreateConsignmentError = (error) => {
  return formatGravityError(error)
}
