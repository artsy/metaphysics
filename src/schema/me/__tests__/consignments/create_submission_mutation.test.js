import { runAuthenticatedQuery } from "test/utils"
import { config as createSubmissionMutation } from "schema/me/consignments/create_submission_mutation.js"

import gql from "test/gql"

describe("UpdateSubmissionMutation", () => {
  it("does not include the id param", () => {
    const mutation = createSubmissionMutation
    expect(Object.keys(mutation.inputFields)).not.toContain("id")
  })

  it("includes the state param", () => {
    const mutation = createSubmissionMutation
    expect(Object.keys(mutation.inputFields)).toContain("state")
  })

  it("updates a submission and returns its new data payload", () => {
    const mutation = gql`
      mutation {
        createConsignmentSubmission(
          input: {
            artist_id: "andy-warhol"
            clientMutationId: "2"
            authenticity_certificate: true
            dimensions_metric: CM
          }
        ) {
          clientMutationId
          consignment_submission {
            artist_id
            authenticity_certificate
            id
            dimensions_metric
          }
        }
      }
    `
    const rootValue = {
      submissionCreateLoader: () =>
        Promise.resolve({
          id: "106",
          artist_id: "andy-warhol",
          authenticity_certificate: true,
          dimensions_metric: "cm",
        }),
    }

    expect.assertions(1)
    return runAuthenticatedQuery(mutation, rootValue).then(data => {
      expect(data).toMatchSnapshot()
    })
  })
})
