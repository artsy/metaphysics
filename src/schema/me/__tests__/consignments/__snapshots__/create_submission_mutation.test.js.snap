// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateSubmissionMutation updates a submission and returns its new data payload 1`] = `
Object {
  "createConsignmentSubmission": Object {
    "clientMutationId": "2",
    "consignment_submission": Object {
      "artist_id": "andy-warhol",
      "authenticity_certificate": true,
      "dimensions_metric": "CM",
      "id": "106",
    },
  },
}
`;
